Homework/src/main/java/io/codelex/javaadvancedtest/exercise1/Card.java

```diff
@@ -0,0 +1,36 @@
+package io.codelex.javaadvancedtest.exercise1;
+
+public abstract class Card {
+
+    private final int number;
+    private final String owner;
+    private final int ccv;
+    protected double balance;
+
+    public Card(int number, String owner, int ccv, double balance) {
+        this.number = number;
+        this.owner = owner;
+        this.ccv = ccv;
+        this.balance = balance;
+    }
+
+    public abstract void addMoney(double amount);
+    public abstract void takeMoney(double amount) throws NotEnoughFundsException;
+    public abstract void displayWarning();
```

Comment:
Tā nav īsti kļūda, bet es laikam to displayWarning metodi iekš abstraktās klases neliktu. Jo lai gan jā - tas piespiež child klasēm implementēt to metodi, bet, ja gadījumā kādai kartei nevajadzētu to warning metodi, tad viņa būtu obligāti jāveido. Bet vari arī atstāt.